### DISCUSSION TITLE
Why the TypeScript migration was reverted?

### DISCUSSION THREAD

**Discussion by mohyddintash** [chanzuckerberg/cellxgene#2579] [Q&A] [answered]
Hey there, first of all, I'd like to thank everyone who contributed to this beautiful piece of work!

I am new to this repo, and was looking around, noticed that there was an attempt to switch to TS and then it was reverted. Can anyone please comment on that as I couldn't trace it from the history context.

Also, I realized there is a repo for the hosted version of cellxgene (that one is in TS 😅), I wonder what's the plan for the two, will they always be kept in sync in terms of functionality, or will one replace the other?

Best,
Mo

**Comment by MaximilianLombardo** [comment, marked as answer]
Hi @mohyddintash - this post was overlooked - my apologies.

> I am new to this repo, and was looking around, noticed that there was an attempt to switch to TS and then it was reverted. Can anyone please comment on that as I couldn't trace it from the history context.

Essentially, there is a bit of historical/organizational context. This is the original repo for the CELLxGENE explorer. During the timeframe that you noted, there was an attempt to migrate to typescript as a part of an engineering excellence goal that was in support of our development of the [CELLxGENE Data Portal](https://cellxgene.cziscience.com/) during its infancy. During this period, the team ended up making the decision to create an alternative version of the explorer that essentially allowed for easier hosting (aka some features are quite costly to replicate in a hosted environment). As a result of that split, this repo was reverted back, while the migration to typescript was completed in the [repo that contains the hosted explorer](https://github.com/chanzuckerberg/single-cell-explorer).

> Also, I realized there is a repo for the hosted version of cellxgene (that one is in TS 😅), I wonder what's the plan for the two, will they always be kept in sync in terms of functionality, or will one replace the other?

These are mostly in sync (with the exception of the custom annotations function being missing from the hosted version). Both are being maintained as they serve different use cases for our users (so no plans for one to replace the other). We are currently prioritizing the development of some additional products in our suite (so no big changes planned for either of the explorer repos in the near future), but we are always open to receiving constructive feedback on the explorer.

**Reply by mohyddintash** [reply]
Thanks @MaximilianLombardo  for taking the time to respond! It makes sense now.
